ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_configs_to_cv.py

```diff
@@ -36,98 +37,136 @@ async def deploy_configs_to_cv(configs: list[CVEosConfig], result: DeployToCvRes
+    Then return the list of existing device container ids. (Always empty if we just created the root container)
+    """
+    root_cv_containers = await cv_client.get_configlet_containers(workspace_id=workspace_id, container_ids=[CONFIGLET_CONTAINER_ID])
+    LOGGER.info("get_or_create_configlet_root_container: Got AVD root container? %s", bool(root_cv_containers))
+    if root_cv_containers:
+        return root_cv_containers[0].child_assignment_ids.values
+
+    # Create the root level container
+    LOGGER.info("deploy_configs_to_cv: Creating AVD root container '%s'", CONFIGLET_CONTAINER_ID)
+    await cv_client.set_configlet_container(
+        workspace_id=workspace_id,
+        container_id=CONFIGLET_CONTAINER_ID,
+        display_name="AVD Configurations",
+        description="Configurations created and uploaded by AVD",
+        query="device:*",
+    )
+    # Add the root level container to the list of root level containers using the studio inputs API (!?!)
+    root_containers: list = await cv_client.get_studio_inputs_with_path(
+        studio_id=STATIC_CONFIGLET_STUDIO_ID, workspace_id=workspace_id, input_path=["configletAssignmentRoots"], default_value=[]
+    )
+    LOGGER.info("deploy_configs_to_cv: Found %s root containers.", len(root_containers))
+    if CONFIGLET_CONTAINER_ID not in root_containers:
+        LOGGER.info("deploy_configs_to_cv: AVD root container not assigned as root container in Static Config Studio. Fixing...")
+        root_containers.append(CONFIGLET_CONTAINER_ID)
+        await cv_client.set_studio_inputs(
+            studio_id=STATIC_CONFIGLET_STUDIO_ID,
+            workspace_id=workspace_id,
+            input_path=["configletAssignmentRoots"],
+            inputs=root_containers,
+        )
+    return []
+
+
+async def deploy_configlet_containers_to_cv(configs: list[CVEosConfig], workspace_id: str, cv_client: CVClient) -> None:
+    """
+    Identify existing containers and ensure they have the correct configuration.
+    Then update/create as needed.
+
+    TODO: Refactor to set_some on supported CV versions
+    """
+    if existing_device_container_ids := await get_existing_device_container_ids_from_root_container(workspace_id, cv_client):
+        existing_device_containers = await cv_client.get_configlet_containers(
+            workspace_id=workspace_id,
+            container_ids=existing_device_container_ids,
+        )
+        LOGGER.info("deploy_configs_to_cv: %s existing device containers under AVD root container.", len(existing_device_containers))
+        # Create dict keyed by container id with value of tuple containing key container parameters. Used later to detect changes.
+        existing_device_containers_by_id = {
+            cv_container.key.configlet_assignment_id: (cv_container.display_name, cv_container.description, cv_container.query, cv_container.configlet_ids)
```

Comment:
`cv_container.configlet_ids` is an instance of dataclass RepeatedString, with attribute `value` with type List[str]
Later in the code this tuple get's compared to manually formed tuple `(display_name, description, query, configlet_ids)` and this always results to False because last item in the tuple on both sides are of a different types (aka class.RepeatedString vs List). This forces this plugin to always re-push all container bindings even if they are already present on CVP side.

Example of `cv_container.configlet_ids`: `RepeatedString(values=['avd-0F2D7FBBBDF42D03B650B2B859C7B3BE'])`
Example of `configlet_ids`: `['avd-0F2D7FBBBDF42D03B650B2B859C7B3BE']`

```suggestion
            cv_container.key.configlet_assignment_id: (cv_container.display_name, cv_container.description, cv_container.query, cv_container.configlet_ids.values)
```